jr-stree/src/main/java/com/fasterxml/jackson/jr/stree/JacksonJrsTreeCodec.java

```diff
@@ -50,6 +65,16 @@ private JrsValue nodeFrom(JsonParser p) throws IOException
             return JrsBoolean.FALSE;
         case JsonTokenId.ID_NUMBER_INT:
         case JsonTokenId.ID_NUMBER_FLOAT:
+            final JsonParser.NumberType n = p.getNumberType();
```

Comment:
Cannot use this method, have a look at jackson-databind `JsonNodeDeserializer` (I think) for better version. Problem being that this method will coerce type because JSON has no native knowledge of optimal FP type to use.